lib/puppet-lint/plugins/check_whitespace/colon_whitespace_before.rb

```diff
@@ -0,0 +1,22 @@
+# Public: Check the manifest tokens for resource type declaration that has no whitespace before a colon and record a warning for each instance found.
+# #
+# https://docs.puppet.com/guides/style_guide.html#spacing-indentation-and-whitespace
+PuppetLint.new_check(:colon_whitespace_before) do
+  def check
+    tokens.each do |token|
+      next if token.prev_token.nil?
+      if (token.type == :COLON && token.prev_token.type == :WHITESPACE)
```

Comment:
Can you add a test for a colon following whitespace that isn't associated with a resource title, maybe in an exec's command string for instance?